vault/router.go

```diff
@@ -62,6 +62,13 @@ func (r *Router) Mount(backend logical.Backend, prefix string, mountEntry *Mount
 	if existing, _, ok := r.root.LongestPrefix(prefix); ok && existing != "" {
 		return fmt.Errorf("cannot mount under existing mount '%s'", existing)
 	}
+	// If this is a secret backend, check to see if the prefix conflicts
+	// with an existing mountpoint
+	if prefix != "" {
+		if match := r.matchingPrefixInternal(prefix); match != "" {
```

Comment:
I'm a little concerned about adding this check in the router.  I've confirmed that adding this check will prevent servers that are in the situation of overlapping mounts from unsealing, where they would start fine before.  This may be ok since vault will never unseal and you can just downgrade to the earlier version, move their mounts, and then proceed with the upgrade.  It's at the very least something to not in the upgrade docs.  @jefferai thoughts?